spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/AggregateApplicationBuilder.java

```diff
@@ -116,15 +117,20 @@ public ConfigurableApplicationContext run(String[] parentArgs) {
 			apps.add(sinkConfigurer);
 		}
 		List<Class<?>> appsToEmbed = new ArrayList<>();
+		String namespace = null;
 		for (int i = 0; i < apps.size(); i++) {
-			appsToEmbed.add(apps.get(i).getApp());
+			AppConfigurer<?> appConfigurer = apps.get(i);
+			appsToEmbed.add(appConfigurer.getApp());
+			if (namespace == null) {
+				namespace = appConfigurer.namespace;
```

Comment:
The goal of the namespace is to disambiguate between channels with the same name (e.g. two `output` channels that belong to successive `Processors`), but that belong to different applications, as they are registered in the `SharedChannelRegistry`. Therefore namespaces should be different for each embedded app.